### ISSUE TITLE
CDCC repeal produces no impact in microsimulation

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/policyengine-us#4044]
Repealing the CDCC by lowering the maximum care expenses to $0 produces no impact, either with [base CPS](https://policyengine.org/us/policy?focus=policyOutput.policyBreakdown&reform=49176&region=us&timePeriod=2024&baseline=2&household=42141) or [enhanced CPS](https://policyengine.org/us/policy?focus=policyOutput.policyBreakdown&reform=49176&region=enhanced_us&timePeriod=2024&baseline=2&household=42141).

It [works at the household level](https://policyengine.org/us/household?focus=householdOutput.earnings&reform=49176&region=enhanced_us&timePeriod=2024&baseline=2&household=42141).

Currently investigating in [this notebook](https://colab.research.google.com/drive/1EQmrMJNLgfx5q6vQZgAOdG94OxbPH5Gl?authuser=0#scrollTo=YNLNtvJl13MP). So far that shows that childcare expenses are OK, but in certain sequences `adjusted_gross_income` sums to zero, resulting in `cdcc_rate` being zero. So this could be a larger issue.

@nikhilwoodruff any chance this could have to do with https://github.com/PolicyEngine/policyengine-core/pull/165?

**Comment by MaxGhenis** [comment]
### TL;DR
This may have resulted from an intersection of uprating, core sequencing, and childcare subsidy issues. We have a fix in #4054 but we need to add testing and improve core to ensure it doesn't affect other variables, and to avoid it recurring.

### More detail
@anth-volk @PavelMakarchuk and I isolated this to https://github.com/PolicyEngine/policyengine-us/pull/3933, which bumped `policyengine_us` from `0.666.1` to `0.667.0` and primarily added uprating factors to Social Security programs to improve accuracy of our long-term projections.

We found that, after that change, running `baseline.calc("cdcc")` produces `NaN` values, but only for 2024, or when run after calculating any other variable for 2024. [These](https://colab.research.google.com/drive/14mTHXJo6f5cvDkquowUTxakm1C8tjxn9?usp=sharing) [notebooks](https://colab.research.google.com/drive/1r9iMEMNhM3HgyWyK5XIigbekTLURuMQw?usp=sharing) show some of our investigation. I filed https://github.com/PolicyEngine/policyengine-core/issues/170 to ensure results do not depend on the sequence of `calc` operations.

We traced the NaNs to the `social_security_dependents` and `social_security_survivors` variables, which the breaking PR newly uprated. It notably did not affect `social_security_retirement` or `social_security_disability`. The distinguishing factor is that retirement and disability are set in cps.py, while the others are not. It appears that empty variables do not get properly uprated to zero and instead go to NaNs; @anth-volk filed https://github.com/PolicyEngine/policyengine-core/issues/169 to address this.

Setting the values to zero in cps.py and removing the uprating factors (probably unnecessary but did it to be safe) fixed that issue, removing NaNs from childcare expenses. However, they were still zero in 2024 (when first invoking other `calc` commands). We found this resulted from us populating `childcare_expenses` in cps.py, a SPM-unit variable that had a formula added in #3813 to accommodate childcare subsidies. By instead creating a new leaf-node variable `spm_unit_pre_subsidy_childcare_expenses` and populating it in cps.py, the CDCC would correctly compute after running other operations. This is increasingly clunky though; we should instead populate a single person-level `pre_subsidy_care_expenses` variable from cps.py (#2900), then aggregate up from there.

I'm going to merge #4054, update the data release, and check the results. But we need further discussion on deeper reliability issues to ensure this caught all instances of this problem and add CI tests to avoid it recurring. @nikhilwoodruff welcome your thoughts here.

**Comment by MaxGhenis** [comment]
Repealing the CDCC now raises $4.1 billion in the base CPS: https://policyengine.org/us/policy?focus=policyOutput.policyBreakdown&reform=49176&region=us&timePeriod=2024&baseline=2&household=42141

$3.6 billion enhanced CPS: https://policyengine.org/us/policy?focus=policyOutput.policyBreakdown&reform=49176&region=enhanced_us&timePeriod=2024&baseline=2&household=42141
